src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -228,6 +228,85 @@ ss::future<bool> seq_writer::write_config(
     });
 }
 
+ss::future<std::optional<bool>> seq_writer::do_delete_config(
+  subject sub, model::offset write_at, seq_writer& seq) {
+    vlog(plog.debug, "delete config sub={} offset={}", sub, write_at);
+
+    try {
+        co_await seq._store.get_compatibility(sub, default_to_global::no);
+    } catch (const exception&) {
+        // subject config already blank
+        co_return false;
+    }
+
+    std::vector<seq_marker> sequences{
+      co_await _store.get_subject_config_written_at(sub)};
+
+    storage::record_batch_builder rb{
+      model::record_batch_type::raft_data, model::offset{0}};
+
+    std::vector<config_key> keys;
+    for (auto s : sequences) {
```

Comment:
Yeah, exactly. This probably explains why I've seen excessive tombstoning for subject versions.

Reply:
I've just tested again;

I added a config at offset 682, 683, then deleted it, got tombstones at 684 for 682, at 685 for 683, and at 686 for 684, added it back at 687, then a delete gave me tombstones for 682 at 688, for 683 at 689, for 687 at 690, and for 688 at 697. So the tombstones seem to be being added to the sequence, rather than deleted from it.

Reply:
Yeah I see what you mean. [This change](https://github.com/redpanda-data/redpanda/pull/13557/commits/8a47a955e3204a45d6d5fc69e012f7a44b7dc07a) (current branch HEAD) seems to fix it if somewhat naively. Look reasonable to you?